src/Client/Updater.php

```diff
@@ -644,6 +640,70 @@ public function download(string $target, ...$extra): PromiseInterface
             });
     }
 
+    /**
+     * Gets a target metadata object that contains the specified target.
+     *
+     * @param string $target
+     *   The path of the target file. Needs to be known to the most recent
+     *   targets metadata downloaded in ::refresh().
+     * @param \Tuf\Metadata\TargetsMetadata|null $targetsMetadata
+     *   The targets metadata to search or null. If null then the search will
+     *   start at the top level 'targets.json' file.
+     * @param string[] $searchedRoles
+     *   The roles that have already been searched.
+     *
+     * @return \Tuf\Metadata\TargetsMetadata|null
+     *   The target metadata with a match for the target, or null no match is
+     *   found.
+     */
+    protected function getMetadataForTarget(string $target, ?TargetsMetadata $targetsMetadata = null, array $searchedRoles = []): ?TargetsMetadata
```

Comment:
Wondering if we should document $searchedRoles as being internal use only, since AFAIK it's used for recursion control. I can't think of any reason why it should be passed by the calling code.

Reply:
Documented this and throwing a UnexpectedValueException if this is not empty and `$targetsMetadata` is empty. This should never be set in a top level call.